modular_zubbers/code/modules/client/verbs/character_directory.dm

```diff
@@ -0,0 +1,218 @@
+	return FALSE
+//Can't believe Bubberstation invented attraction and gender in the year December 2023
+
+//Add a cooldown for the character directory to the client, primarily to stop server lag from refresh spam
+/client
+	COOLDOWN_DECLARE(char_directory_cooldown)
+
+//Make a verb to open the character directory
+/client/verb/show_character_directory()
+	set name = "Character Directory"
+	set category = "OOC"
+	set desc = "Shows a listing of all active characters, along with their associated OOC notes, flavor text, and more."
+
+	// This is primarily to stop malicious users from trying to lag the server by spamming this verb
+	if(!COOLDOWN_FINISHED(src, char_directory_cooldown))
+		to_chat(src, span_alert("Hold your horses! Its still refreshing!"))
+		return
+	COOLDOWN_START(src, char_directory_cooldown, 10)
+
+//Check if there's not already a character directory open; open a new one if one is not present
+	if(!GLOB.character_directory)
+		GLOB.character_directory = new
+	GLOB.character_directory.ui_interact(mob)
+
+// This is a global singleton. Keep in mind that all operations should occur on user, not src.
+/datum/character_directory
+
+/datum/character_directory/ui_state(mob/user)
+	return GLOB.always_state
+
+/datum/character_directory/ui_interact(mob/user, datum/tgui/ui)
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "ZubbersCharacterDirectory", "Character Directory")
+		ui.open()
+
+//We want this information to update any time the player updates their preferences, not just when the panel is refreshed
+/datum/character_directory/ui_data(mob/user)
+	. = ..()
+	var/list/data = .
+
+//Collect the user's own preferences for the top of the UI
+	if (user?.client?.prefs)
+		data["personalVisibility"] = READ_PREFS(user, toggle/show_in_directory)
+		data["personalAttraction"] = READ_PREFS(user, choiced/attraction)
+		data["personalGender"] = READ_PREFS(user, choiced/display_gender)
+		data["personalErpTag"] = READ_PREFS(user, choiced/erp_status)
+		data["personalVoreTag"] = READ_PREFS(user, choiced/erp_status_v)
+		data["personalHypnoTag"] = READ_PREFS(user, choiced/erp_status_hypno)
+		data["personalNonconTag"] = READ_PREFS(user, choiced/erp_status_nc)
+		data["prefsOnly"] = TRUE
+
+	data["canOrbit"] = isobserver(user)
+
+	return data
+
+/datum/character_directory/ui_static_data(mob/user)
+	. = ..()
+	var/list/data = .
+
+	var/list/directory_mobs = list()
+	//We want the directory to display only alive players, not observers or people in the lobby
+	for(var/mob/mob in GLOB.alive_player_list)
+		// These are the variables we're trying to display in the directory
+		var/name = null
+		var/species = null
+		var/ooc_notes = ""
+		var/flavor_text = ""
+		var/attraction = null
+		var/gender = null
+		var/erp = null
+		var/vore = null
+		var/hypno = null
+		var/noncon = null
+		var/character_ad = ""
+		var/exploitable = ""
+		var/ref = REF(mob)
+		//Just in case something we get is not a mob
+		if(!mob)
+			continue
```

Comment:
Is this necessary? (Also weakly typed languages were a mistake...)

Reply:
Not really, it's very unlikely that something we get is not a mob, but I think linters got angry when I removed it so